template/thymeleaf/src/main/java/com/blazebit/template/thymeleaf/ThymeleafTemplateProcessor.java

```diff
@@ -51,8 +50,9 @@ public class ThymeleafTemplateProcessor implements TemplateProcessor<String>, Se
      * @param configurationSource The configuration source
      */
     public ThymeleafTemplateProcessor(ConfigurationSource configurationSource) {
-        this.templateEngine = configurationSource.getPropertyOrDefault(THYMELEAF_TEMPLATE_ENGINE_PROPERTY, ITemplateEngine.class, val -> null, val -> new TemplateEngine());
-        this.templateName = configurationSource.getPropertyOrFail(THYMELEAF_TEMPLATE_NAME_PROPERTY, String.class, Function.identity());
+        ServiceProvider serviceProvider = configurationSource.getPropertyOrDefault(SERVICE_PROVIDER_PROPERTY, ServiceProvider.class, val -> null, (serviceType) -> null);
+        this.templateEngine = serviceProvider.getService(ITemplateEngine.class).orElse(new TemplateEngine());
```

Comment:
Looks like NPEs are possible. Please throw some meaningful exception i.e. `Required ServiceProvider is not available in configuration source`. Alternatively, try to handle the `null` service provider maybe?